torchci/lib/bot/cancelWorkflowsOnCloseBot.ts

```diff
@@ -0,0 +1,64 @@
+import { Probot } from "probot";
+
+function cancelWorkflowsOnCloseBot(app: Probot): void {
+  app.on("pull_request.closed", async (ctx) => {
+    const owner = ctx.payload.repository.owner.login;
+    const repo = ctx.payload.repository.name;
+    const prNumber = ctx.payload.pull_request.number;
+    const senderLogin = ctx.payload.sender.login;
+    const senderId = ctx.payload.sender.id;
+
+    if (
+      senderLogin == "pytorchmergebot" ||
+      senderLogin == "pytorchbot" ||
+      senderLogin == "pytorch-bot[bot]" ||
+      senderId == 97764156 || // pytorchmergebot's id
+      senderId == 54816060 || // pytorch-bot's id
+      senderId == 21957446 // pytorchbot's id
+    ) {
+      return;
+    }
+
+    const graphqlquery = `query ($owner: String!, $repo: String!, $prNumber: Int!) {
```

Comment:
Have you tried https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository?  Some of its parameter, `head_sha`, `branch`, and `status` seems relevant

Reply:
Oh thats so much better, thanks